test/parallel/test-cli-eval.js

```diff
@@ -113,3 +113,16 @@ child.exec(nodejs + ` -e 'require("child_process").fork("${emptyFile}")'`,
   assert.strictEqual(proc.stderr, '');
   assert.strictEqual(proc.stdout, 'start\nbeforeExit\nexit\n');
 }
+
+[ '-arg1',
+  '-arg1 arg2 --arg3',
+  '--',
+].forEach(function(args) {
+  const evalopt = ' --eval "console.log(process.argv.slice(1).join(\' \'))"';
+  const cmd = `${nodejs}${evalopt} -- ${args}`;
+  child.exec(cmd, common.mustCall(function(err, stdout, stderr) {
+    assert.strictEqual(stdout, args + '\n');
+    assert.strictEqual(stderr, '');
+    assert.strictEqual(err, null);
+  }));
+});
```

Comment:
passed in, so that the script.js can use it to indicate the end of its options:

`node --some-node-option script.js --some-script-option -- --not-a-script-option`